src/basic/hostname-util.h

```diff
@@ -14,10 +14,14 @@ char* gethostname_short_malloc(void);
 int gethostname_strict(char **ret);
 
 bool valid_ldh_char(char c) _const_;
-bool hostname_is_valid(const char *s, bool allow_trailing_dot) _pure_;
-char* hostname_cleanup(char *s);
 
-#define machine_name_is_valid(s) hostname_is_valid(s, false)
+typedef enum ValidHostnameFlags {
+        VALID_HOSTNAME_TRAILING_DOT = 1 << 0,   /* Accept trailing dot on multi-label names */
+        VALID_HOSTNAME_DOT_HOST     = 1 << 1,   /* Accept ".host" as valid hostname */
```

Comment:
Not that it matters much, but `VALID_HOSTNAME_TRAILING_DOT` and `VALID_HOSTNAME_LEADING_DOT` would be a  more consistent way of naming these flags.

Reply:
no. it's not the trailing dot we accept. it's literally ".host" we accept as reference for the local system. i.e. the 5 characters: dot, "h", "o", "s", "t".

Reply:
I see. Sorry for the noise.